source/slang/slang-emit-c-like.cpp

```diff
@@ -217,8 +217,34 @@ void CLikeSourceEmitter::emitSimpleType(IRType* type)
         outNumThreads[i] = decor ? Int(getIntVal(decor->getOperand(i))) : 1;
     }
     return decor;
-}
-
+}
+
+List<IRWitnessTableEntry*> CLikeSourceEmitter::getSortedWitnessTableEntries(IRWitnessTable* witnessTable)
+{
+    List<IRWitnessTableEntry*> sortedWitnessTableEntries;
+    auto interfaceType = cast<IRInterfaceType>(witnessTable->getOperand(0));
+    auto witnessTableItems = witnessTable->getChildren();
+    for (UInt i = 0; i < interfaceType->getOperandCount(); i++)
```

Comment:
For now I think the simple code is warranted, but this logic is yet another case of the Slang compiler being "accidentally quadratic" in ways that could bite us later. In this case, the performance of `getSortedWitnessTableEntries` is going to be quadratic in the number of requirements/entries.

We can obviously fix this down the line if the performance becomes an issue, but it might be worth making a note of the lurking performance problem and a  basic idea of how to resolve it at the cost of more complicated code.